frontend/src/components/eceweApplication/EceweEligibilityQuestions.vue

```diff
@@ -0,0 +1,519 @@
+    ]),
+    ...mapState(useApplicationStore, [
+      'formattedProgramYear',
+      'programYearId',
+      'applicationStatus',
+      'unlockEcewe',
+    ]),
+    ...mapState(useOrganizationStore, ['organizationProviderType']),
+    ...mapState(useReportChangesStore, ['isEceweUnlocked', 'changeRequestStatus']),
+    showCSSEAWarning(){
+      //this is only for 2025-26
+      return this.model?.publicSector === ECEWE_IS_PUBLIC_SECTOR_EMPLOYER.NO && this.model?.describeOrgCSSEA === ECEWE_DESCRIBE_ORG_TYPES.MEMBER_OF_CSSEA
+    },
+    showApplicableSectorQuestion() {
+      //This question is only valid from 2023-24 and before.
+      //The template manages 2025 onward so it should only get called for historical applications
+      return  this.model.belongsToUnion === 1 && this.model.optInECEWE === ECEWE_OPT_IN_TYPES.OPT_IN
+    },
+    showConfirmationQuestion() {
+      //This question is only valid from 2023-24 and before.
+      return this.model.applicableSector === ECEWE_SECTOR_TYPES.OTHER_UNION && this.model.belongsToUnion === 1 && this.model.optInECEWE === ECEWE_OPT_IN_TYPES.OPT_IN
+
+    },
+    showFundingModelQuestion() {
+      return (
+        this.model.applicableSector === ECEWE_SECTOR_TYPES.CSSEA &&
+        this.model.belongsToUnion === 1 &&
+        this.model.optInECEWE === ECEWE_OPT_IN_TYPES.OPT_IN
+      );
+    },
+    showJJEPQuestion() {
+      return (
+        this.model.fundingModel === this.fundingModelTypeList[1].id ||
+        this.model.fundingModel === this.fundingModelTypeList[2].id
+      );
+    },
+    languageYearLabel() {
+      return this.getLanguageYearLabel;
+    },
+    programYearTypes() {
+      return PROGRAM_YEAR_LANGUAGE_TYPES;
+    },
+  },
+  created() {
+    this.ORGANIZATION_PROVIDER_TYPES = ORGANIZATION_PROVIDER_TYPES;
+    this.ECEWE_OPT_IN_TYPES = ECEWE_OPT_IN_TYPES
+    this.ECEWE_SECTOR_TYPES = ECEWE_SECTOR_TYPES
+    this.ECEWE_DESCRIBE_ORG_TYPES = ECEWE_DESCRIBE_ORG_TYPES
+    this.ECEWE_IS_PUBLIC_SECTOR_EMPLOYER = ECEWE_IS_PUBLIC_SECTOR_EMPLOYER
+    this.ECEWE_UNION_AGREEMENT_REACHED = ECEWE_UNION_AGREEMENT_REACHED
+    this.model = { ...this.eceweModel };
+  },
+
+  methods: {
+    getFormData() {
+      return this.model;
+    },
+
+    //For change requests - if a facility has previously opted-in or said yes to having a union on their CORE application
+    //they are not allowed to change that response, so question becomes read only.
+    //if not a change request- default to the prop calculated by the parent.
+    isQuestionReadOnly(question) {
+      if (this.isChangeRequest) {
+        if (this.isEceweUnlocked || !this.changeRequestStatus || this.changeRequestStatus === 'INCOMPLETE') {
+          return (
+            (question === 'optInECEWE' && this.optinECEWEChangeRequestReadonly) ||
+            (question === 'belongsToUnion' && this.belongsToUnionChangeRequestReadonly)
+          );
+        } else if (this.changeRequestStatus !== 'INCOMPLETE') {
```

Comment:
I was going to suggest a constant here but I think that's a rabbit hole you do not want to go down!

Reply:
untangling the readOnly code was already very difficult so I'm happy to leave it :)